### PR TITLE
Modify the documentation of nmcli: conn_name

### LINKED ISSUE #29302: nmcli: conn_name required
From @maltefiala on 2016-04-17T08:55:33Z
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

nmcli
##### ANSIBLE VERSION

```
# latest head of 2016-04-16
ansible 2.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

CentOS Linux release 7.2.1511 (Core) 
##### SUMMARY

Conn Name Required althouth stated otherwise in documentation:

> Where conn_name will be the name used to call the connection. **when not provided a default name is generated: `<type>[-<ifname>][-<num>]`**
##### STEPS TO REPRODUCE

```
- name: Configure network adapter
  nmcli: type=ethernet ip4="192.168.1.1/16" ifname="enp0s1" autoconnect=yes state=present
```
##### EXPECTED RESULTS

OK
##### ACTUAL RESULTS

```
FAILED! => {"changed": false, "failed": true, "msg": "missing required arguments: conn_name"}
```

Copied from original issue: ansible/ansible-modules-extras#2044

### PR DESCRIPTION
**Pull request by amolkahat**
##### SUMMARY
Modify the documentation of nmcli: conn_name
Fixes: #29302

Signed-off-by: Amol Kahat <akahat@redhat.com>

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
nmcli

##### ANSIBLE VERSION
```
 $ ansible --version
ansible 2.5.0
  config file = None
  configured module search path = [u'/home/admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Jan 17 2018, 14:28:32) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
(ansible_venv) 
```

### CODE REVIEW

File: lib/ansible/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
     conn_name:
         required: True
         description:
-            - 'Where conn_name will be the name used to call the connection. when not provided a default name is generated: <type>[-<ifname>][-<num>]'
+            - 'Where conn_name will be the name used to call the connection. <type>[-<ifname>][-<num>]'
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nmcli.py]: s/.../Where conn_name will be the name used to call the connection./

**Comment by amolkahat** [comment]
@maxamillion Could you please review it?

**Comment by Akasurde** [comment]
@amolkahat Could you please rebase this branch and we can take it forward ?

**Comment by acozine** [comment]
Thanks @amolkahat for finding this error and opening a PR to fix it! I apologize that your PR got lost in the shuffle this winter/spring. The change you wanted (documenting the fact that `conn_name` is a required field) was added by #36909, so I'm closing this PR and issue #29302 also.

**Comment by amolkahat** [comment]
@acozine Thansk :+1:

**Pull request closed without merging**